client/apps/settings/client/curations/display/components/canvas_container.jsx

```diff
@@ -0,0 +1,53 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import Canvas from './canvas.jsx'
+
+export default class CanvasContainer extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      activeLayout: props.campaign.canvas.layout || 'overlay'
+    }
+  }
+
+  setActiveLayout = (layout) => {
+    this.setState({activeLayout: layout})
+    this.props.onChange('canvas.layout', layout, this.props.index)
```

Comment:
What happens if you switch from a video layout to overlay? Or slideshow with many images to video? Maybe some level of validation on asset type and quantity could be worthwhile here.

Reply:
If you switch from a slideshow with many images to another layout it does not delete them, and only shows the new one.  

Video is not allowed to be uploaded unless you are a video section, but it does not delete the files -- was planning to add a warning for not-allowed files but in a separate PR. 

Slideshows limit uploads to 5 images, and hide the upload for more if you have 5.

Reply:
Yeah it's good that the individual layouts have some validation. For the pilot it's not necessary but I thought it could spark some ideas :). API rules are little more relaxed with Curations, but ideally we could have "layout rules" that handle switching more gracefully -- ie, mapping business rules like "canvas/video has only 1 video asset" to the API so the client doesn't have to think about the possibility of multiple assets or different types.

Reply:
Yeah I figured there would be some ability to switch between layouts while you are building, so it seems like we don't want to prevent saving.  However, yeah we do need to validate.  Was planning to do something like image_collections, where an alert tells you if you have too many assets.